pebblo/app/service/prompt/prompt_service.py

```diff
@@ -0,0 +1,241 @@
+            )
+            if not app_exists and ai_app_obj:
+                message = f"{self.app_name} app doesn't exists"
+                logger.error(message)
+                return self._return_response(message=message, status_code=500)
+            if ai_app_obj and len(ai_app_obj) > 0:
+                ai_app_data = ai_app_obj[0]
+            else:
+                message = f"{self.app_name} app doesn't exists"
+                logger.error(message)
+                return self._return_response(message=message, status_code=500)
+
+            document_accessed = []
+            for data in retrieval_data["context"]:
+                doc_name = data.get("retrieved_from")
+                if doc_name not in document_accessed:
+                    document_accessed.append(doc_name)
+
+            # Update entry in AiUser if exists else create
+            if retrieval_data.get("user"):
+                _, ai_user = self.db.query(
+                    table_obj=AiUser, filter_query={"name": retrieval_data["user"]}
+                )
+                if ai_user and len(ai_user) > 0:
+                    ai_user = ai_user[0]
+                    user_id = ai_user.data.get("id")
+                    retrieval_data["user"] = ai_user.data.get("name")
+                    retrieval_data["user_id"] = user_id
+                    existing_document_accessed = ai_user.data.get(
+                        "documentsAccessed", []
+                    )
+                    for doc_name in document_accessed:
+                        if doc_name not in existing_document_accessed:
+                            existing_document_accessed.append(doc_name)
+                    ai_user.data["documentsAccessed"] = existing_document_accessed
+                    status, message = self.db.update_data(
+                        table_obj=ai_user, data=ai_user.data
+                    )
+                    if not status:
+                        logger.error(f"Failed during updating AiUser: {message}")
+                        return self._return_response(message=message, status_code=500)
+                else:
+                    # Initialize Variables
+                    current_time = self._get_current_datetime()
+                    last_used = current_time
+                    metadata = Metadata(createdAt=current_time, modifiedAt=current_time)
+                    ai_user_obj = aiuser(
+                        name=retrieval_data["user"],
+                        metadata=metadata,
+                        userAuthGroup=retrieval_data.get("linked_groups", []),
+                        documentsAccessed=document_accessed,
+                        lastUsed=last_used,
+                    )
+                    insert_status, entry = self.db.insert_data(
+                        AiUser, ai_user_obj.dict()
+                    )
+                    if insert_status:
+                        logger.debug(f"Entry: {entry} in AiUser completed")
```

Comment:
Do we need to log entry ?

Reply:
@dristysrivastava Can you please check this?